grakn-core/src/main/java/ai/grakn/graql/VarPattern.java

```diff
@@ -180,6 +180,20 @@
     @CheckReturnValue
     VarPattern relates(VarPattern type);
 
+    /**
+     * @param roleType a {@link Role} id that this relation type variable must have
+     * @return this
+     */
+    @CheckReturnValue
+    VarPattern relates(String roleType, String superRoleType);
```

Comment:
Since the argument can be null, we can say: `..., @Nullable String superRoleType);`

(and same for below)